### DISCUSSION TITLE
Overwrite existing binaries with cask installations

### DISCUSSION THREAD

**Discussion by kouwei32** [Homebrew/discussions#2909] [Casks]
`Error: It seems there is already a Binary at '/usr/local/bin/dftest'.`

I'm currently trying to migrate some specific applications over to cask mainly so I don't have to bother too much with manual updates. `wireshark` specifically seems to be giving me some trouble, as the cask installation automatically fails when it runs into some ancient binaries (probably from the older installation that were removed). `--force` in its infinite wisdom decides that overwriting binaries is bad, and there doesn't seem to be any options to just ignore existing binaries or list them for deletion. I'm thinking maybe a dry run could work with `grep` to find all the conflicting files, but not sure how I could execute that.

Any ideas on telling Homebrew to overwrite binaries? Couldn't find anything suitable in the man page.

**Comment by gromgit** [comment]
If I understand you correctly, you're trying to replace manually-installed Mac apps with their corresponding casks. In that case, it's generally a good idea to uninstall the app first before cask-installing it again, otherwise you may find old executables (read: security concerns) that aren't removed by the new cask, simply because they aren't in the new cask.

If there isn't an official uninstaller for the app concerned, I recommend the free [AppCleaner](https://freemacsoft.net/appcleaner/) app. I've used it for years, I've yet to find an installed app that confuses it, and it hunts down stuff like kexts that may not be removed by a simple `rm -fr /Applications/App_To_Be_Casked.app`.

And, of course, it's been casked too: `brew install --cask appcleaner`.

**Reply by kouwei32** [reply]
AppCleaner doesn't seem to recognize the binaries in `/usr/local/bin` and only picks up on 2 irrelevant empty folders. Can't tell which binaries are left over still. `brew` just stops running if it hits one of them so...

**Reply by neutric** [reply]
The 3959 currently available casks share 567 binary installations, or ~0,143 per cask. Wireshark is an outlier, sure, but this smells like over-engineering a one-off thing. Once Homebrew handles updates, the binaries will be updated no problem. I'd just manually rm the few conflicting binaries and move on.

**Reply by gromgit** [reply]
> I'd just manually rm the few conflicting binaries and move on.

Given @kouwei32's followup comment, I'd have to agree, but recommend `ls -l /usr/local/bin/dfstat` first to see if it's a symbolic link to something recognizable.
